### PR TITLE
docs: add tfplugindocs and properties description

### LINKED ISSUE #125: Implement terraform-plugin-docs

### PR DESCRIPTION
**Pull request by ianaya89**
## Affected Components
* [ ] Resources
* [ ] Tests
* [x] Docs
* [ ] Other

## Style
* [ ] Terraform code is formatted with `terraform fmt`
* [ ] Go code is formatted with `go fmt`

## Notes for the Reviewer
- Create examples directory
- Add description to properties
- Add tfdocsplugin and make command

> Resolves #125

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -1,68 +1,51 @@
...
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "checkly Provider"
+subcategory: ""
+description: |-
+  
+---
 
-You can find a quick [step-by-step guide](https://www.checklyhq.com/docs/integrations/terraform/) in Checkly's documentation.
+# checkly Provider
```
Comment (ragog) [inline docs/index.md:9]: @ianaya89 I see this is generated. Can we still get Checkly capitalised?

File: docs/index.md [hunk 2]
```diff
@@ -1,68 +1,51 @@
...
 
-# specify the Checkly provider
 terraform {
+  required_version = "= 1.1.17"
   required_providers {
     checkly = {
-      source = "checkly/checkly"
-      version = "1.3.0"
+      source  = "checkly/checkly"
+      version = "0.0.0-canary"
```
Comment (ragog) [inline docs/index.md:23]: @ianaya89 not sure why this is pointing to this version.

**Review by ragog — APPROVED** [review]

File: checkly/resource_check.go [hunk 3]
```diff
@@ -52,34 +55,41 @@ func resourceCheck() *schema.Resource {
...
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Determines if any notifications will be sent out when a check fails and/or recovers.",
 			},
 			"should_fail": {
-				Type:     schema.TypeBool,
-				Optional: true,
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Allows to invert the behaviour of when a check is considered to fail. Allows for validating error status like 404.",
```
[the remark this answers is not in the document]
Comment (ianaya89) [reply checkly/resource_check.go:78]: Sounds good to me, any suggestion on what to put?

File: checkly/resource_check.go [hunk 4]
```diff
@@ -296,8 +327,9 @@ func resourceCheck() *schema.Resource {
 							},
 						},
 						"body": {
-							Type:     schema.TypeString,
-							Optional: true,
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "Possible values `NONE`, `JSON`, `FORM`, `RAW`, and `GRAPHQL`.",
```
[the remark this answers is not in the document]
Comment (ianaya89) [reply checkly/resource_check.go:332]: Sorry, didn't get you

File: docs/index.md (hunk 1 above)
Comment (ianaya89) [reply docs/index.md:9]: Not sure how, but will take a look!

**Review by ragog — DISMISSED** [review]
LGTM, let's merge these first changes and take a deeper look at attribute descriptions in a separate PR

**Pull request merged**